examples/nivisa_dmm_measurement/_visa_dmm.py

```diff
@@ -0,0 +1,136 @@
+        self,
+        resource_name: str,
+        id_query: bool = True,
+        reset_device: bool = True,
+        simulate: bool = False,
+    ) -> None:
+        """Open NI-VISA DMM session."""
+        # Create a real or simulated VISA resource manager."""
+        visa_library = f"{_SIMULATION_YAML_PATH}@sim" if simulate else ""
+        resource_manager = pyvisa.ResourceManager(visa_library)
+
+        session = resource_manager.open_resource(
+            resource_name, read_termination="\n", write_termination="\n"
+        )
+
+        if not isinstance(session, pyvisa.resources.MessageBasedResource):
+            raise TypeError("The 'session' object must be an instance of MessageBasedResource.")
+        self._session = session
+
+        if id_query:
+            self._validate_id()
+
+        if reset_device:
+            self._reset()
+
+    def close(self) -> None:
+        """Close the session."""
+        self._session.close()
+
+    def __enter__(self) -> Self:
+        """Context management protocol. Returns self."""
+        return self
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_val: Optional[BaseException],
+        traceback: Optional[TracebackType],
+    ) -> None:
+        """Context management protocol. Calls close()."""
+        self.close()
+
+    def configure_measurement_digits(
+        self, function: Function, range: float, resolution_digits: float
+    ) -> None:
+        """Configure the common properties of the measurement.
+
+        These properties include function, range, and resolution_digits.
+        """
+        function_enum = _FUNCTION_TO_VALUE[function]
+        resolution_value = _RESOLUTION_DIGITS_TO_VALUE[str(resolution_digits)]
+
+        self._session.write("CONF:%s %.g,%.g" % (function_enum, range, resolution_value))
+        self._check_error()
+
+    def read(self) -> float:
+        """Acquires a single measurement and returns the measured value."""
+        response = self._session.query("READ?")
+        self._check_error()
+        return float(response)
+
+    def _check_error(self) -> None:
+        """Query the instrument's error queue."""
+        response = self._session.query("SYST:ERR?")
+        fields = response.split(",", maxsplit=1)
+        assert len(fields) >= 1
+        if int(fields[0]) != 0:
+            raise RuntimeError("Instrument returned error %s: %s" % (fields[0], fields[1]))
+
+    def _validate_id(self) -> None:
+        """Check the selected instrument is proper and responding.."""
+        instrument_id = self._session.query("*IDN?")
+        if not any(id_check in instrument_id for id_check in _SUPPORTED_INSTRUMENT_IDS):
+            raise RuntimeError(
+                "The ID query failed. This may mean that you selected the wrong instrument, your instrument did not respond, or you are using a model that is not officially supported by this driver. Instrument ID: %s"
+                % (instrument_id)
```

Comment:
Split long line and use an f-string
```suggestion
                "The ID query failed. This may mean that you selected the wrong instrument, your instrument did not respond, "
                f"or you are using a model that is not officially supported by this driver. Instrument ID: {instrument_id}"
```